spring-cloud-app-broker-core/src/main/java/org/springframework/cloud/appbroker/workflow/instance/AppDeploymentDeleteServiceInstanceWorkflow.java

```diff
@@ -73,14 +75,8 @@ public AppDeploymentDeleteServiceInstanceWorkflow(BrokeredServices brokeredServi
 	}
 
 	private Flux<String> deleteBackingServices(DeleteServiceInstanceRequest request) {
-		return backingAppManagementService.getDeployedBackingApplications(request.getServiceInstanceId())
-			.flatMapMany(Flux::fromIterable)
-			.flatMap(backingApplication ->
-				Flux.fromIterable(backingApplication.getServices())
-					.map(servicesSpec -> BackingService.builder()
-						.serviceInstanceName(servicesSpec.getServiceInstanceName())
-						.build())
-					.collectList())
+		return collectConfiguredBackingServices(request)
+			.mergeWith(collectBoundBackingServices(request))
```

Comment:
Would there be duplicate BackingServices among those returned by `collectConfiguredBackingServices` and `collectBoundBackingServices` ?

Does this lead to polluting traces when trying to delete twice a service instance ?

Reply:
Thanks for the feedback. Yes, it would result in multiple delete requests if a backing service instance is bound to an app. I'll consider how to eliminate those duplicates.

Reply:
I've added a new commit that addresses the issue of duplicate service instances